packages/pluggableWidgets/datagrid-web/src/components/Header.tsx

```diff
@@ -49,31 +49,55 @@ export function Header<D extends object>({
             {...rest}
             style={{
                 ...style,
-                ...(!resizable ? { flex: "1 1 0px" } : {}),
-                ...(!sortable || !column.canSort ? { cursor: "unset" } : {})
+                ...(!props.resizable ? { flex: "1 1 0px" } : {}),
+                ...(!props.sortable || !props.column.canSort ? { cursor: "unset" } : {})
             }}
+            title={props.column.render("Header") as string}
         >
             <div
-                id={column.id}
-                className={classNames("column-container", canDrag && column.id === dragOver ? "dragging" : "")}
+                id={props.column.id}
+                className={classNames(
+                    "column-container",
+                    canDrag && props.column.id === props.dragOver ? "dragging" : ""
+                )}
                 {...draggableProps}
             >
                 <div
-                    id={column.id}
+                    id={props.column.id}
                     className={classNames("column-header", canSort ? "clickable" : "")}
-                    onClick={canSort ? onClick : undefined}
+                    onClick={
+                        canSort
+                            ? e => {
+                                  /**
+                                   * Always analyse previous values to predict the next
```

Comment:
Basically we have 3 states, Ascending, Descending and unsorted. And the state machine will look like
Unsorted => Asc => Desc => Unsorted
So if we assume unsorted is the beginning, you wont need the `&& props.column.isSortedDesc === undefined` because that situation wont occur :)

Reply:
Done